interfaces/IWatermark.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "Module.h"
+
+namespace WPEFramework {
+namespace Exchange {
+
+    struct EXTERNAL IWatermark : virtual public Core::IUnknown {
+        enum { ID = ID_WATERMARK };
+
+        virtual ~IWatermark() {}
+        virtual uint32_t Initialize(string waylandDisplay) = 0;
+        virtual uint32_t Deinitialize() = 0;
+        virtual bool ShowWatermark(const bool show) = 0;
+        virtual bool CreateWatermark(uint32_t id, uint32_t zorder) = 0;
+        virtual bool UpdateWatermark(uint32_t id, uint32_t key, uint32_t size) = 0;
+        virtual bool AdjustWatermark(uint32_t id, uint32_t zorder) = 0;
+        virtual bool DeleteWatermark(uint32_t id) = 0;
+    };
```

Comment:
I think there is a tautology in the naming. The interface is already called IWatermark so do not repeat it in all the methods.
Also I do not know the design but I think that the Initialize and Deinitalize are not related to the Watermark functionality. Hence why I would not expect them here. Since IUnknown supports lifetime management I would expect this functionality to be part of the implementatin and hence when you instantiate it, you apply these methods on the implementation but they do not need expeosure on the interface.
Also missing the constness, an interface shoul, without tomuch explenation show the intention. I think that the id is something you feed into the methods and thus could all be const, same goed for zorder, key ans ssize I think.... if it is expected to be changesd, it must be at least a reference.
Why push a full string to the Initialize? Would a const string& not be sufficient ?